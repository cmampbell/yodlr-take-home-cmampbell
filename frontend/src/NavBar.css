.NavBar {
    height: 8vh;
    width: 100%;
    display: flex;
    flex-direction: row;
    justify-content: flex-end;
    align-items: baseline;
    margin-top: 15px;
}

#NavBar-Yodlr{
    position: absolute;
    left: 3vw;
    padding-left: 5px;
    padding-right: 5px;
    border-style: solid;
}

.NavBar a {
    color: black;
    text-decoration: none;
    text-transform: uppercase;
    font-weight: bolder;
    font-size: 1.5rem;
    margin-right: 1.5rem;
    transition: color .2s;
}

.NavBar a:hover:not(#NavBar-Yodlr) {
    color: rgb(22, 22, 92);
}

.NavBar a.active:not(#NavBar-Yodlr){
    color: gray;
}