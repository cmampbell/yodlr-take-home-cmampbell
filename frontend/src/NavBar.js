import React from 'react';
import { NavLink } from 'react-router-dom'
import "./NavBar.css"

function App() {
  return (
    <div className="NavBar">
      {/* Move links into top right corner */}
      <NavLink to="/" id="NavBar-Yodlr">Yodlr</NavLink>
      <NavLink to="/signup">Sign Up</NavLink>
      <NavLink to="/admin">Admin Page</NavLink>
      {/* All other components will be rendered here */}
    </div>
  );
}

export default App;