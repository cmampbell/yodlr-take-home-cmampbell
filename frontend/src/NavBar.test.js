import React from "react";
import renderWithRouter from "./renderWithRouter";
import NavBar from "./NavBar";


describe('NavBar tests', () => {
    it('should render without crashing', () => {
        // define a full route as the first argument
        renderWithRouter(<NavBar />)
    })

    it('should match the snapshot', ()=> {
        const { asFragment } = renderWithRouter(<NavBar />)

        expect(asFragment()).toMatchSnapshot();
    })
})