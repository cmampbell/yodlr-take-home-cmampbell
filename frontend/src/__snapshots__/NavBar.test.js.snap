// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar tests should match the snapshot 1`] = `
<DocumentFragment>
  <div
    class="NavBar"
  >
    <a
      aria-current="page"
      class="active"
      href="/"
      id="NavBar-Yodlr"
    >
      Yodlr
    </a>
    <a
      class=""
      href="/signup"
    >
      Sign Up
    </a>
    <a
      class=""
      href="/admin"
    >
      Admin Page
    </a>
  </div>
</DocumentFragment>
`;
